### ISSUE TITLE
Dota 2 segfaults on startup using the new Sniper runtime

### ISSUE THREAD

**Issue by SkyLeite** [ValveSoftware/steam-runtime#604]
#### Your system information

* Steam Runtime Version: 
* Distribution (e.g. Ubuntu 18.04): NixOS
* Link to your full system information (Help -> System Information) in a [Gist](https://gist.github.com/): https://gist.github.com/SkyLeite/6d20f24fd8e291e4f8be95773c91bebe
  
* Have you checked for system updates?: [Yes/No] Yes
* What compatibility tool are you using?: [None / Steam Linux Runtime / Proton 5.13+ / older Proton] Steam Linux Runtime
* If you are using Steam Linux Runtime for native Linux games: What versions are listed in `SteamLinuxRuntime/VERSIONS.txt`?
#Name	Version		Runtime	Runtime_Version	Comment
depot	0.20230605.51441			# Overall version number
pressure-vessel	0.20230605.0	scout		# pressure-vessel-bin.tar.gz
scripts	0.20230605.0			# from steam-runtime-tools
sniper	0.20230605.51441	sniper	0.20230605.51441	# sniper_platform_0.20230605.51441/

* If you are using Steam Linux Runtime, or Proton 5.13 or newer: What versions are listed in `SteamLinuxRuntime_soldier/VERSIONS.txt`?

#### Please describe your issue in as much detail as possible:
Launching Dota crashes on startup before any window is created.

Output of `SteamLinuxRuntime_sniper/var/slr-latest.log`: https://gist.github.com/SkyLeite/70888ed789ec8e6086903b930533cbb5

#### Steps for reproducing this issue:

1. Install Dota 2
2. Click Play

**Comment by smcv** [comment]
Please could a Valve developer look up what's going on in `CrashID=bp-b254d41a-fe8b-449a-ab0e-d49732230807`?

**Comment by smcv** [comment]
It would also be useful if you can try the `client_beta` branch of "Steam Linux Runtime - sniper". It's the same procedure as [opting in to the beta branch of a game](https://help.steampowered.com/en/faqs/view/5A86-0DF4-C59E-8C4A), but instead of editing the Properties of the game, you'd edit the Properties of the "Steam Linux Runtime - sniper" tool.

**Comment by smcv** [comment]
Another useful thing to try would be launching a different title that uses sniper. At the moment there are not very many, so that basically means [Endless Sky](https://store.steampowered.com/app/404410/Endless_Sky/) or [Retroarch](https://store.steampowered.com/app/1118310/RetroArch/), which are both free, open source and not particularly big.

If those work but Dota 2 doesn't, then the crash is probably more likely to be a Dota 2 bug.

**Comment by smcv** [comment]
From your log, nothing jumps out as particularly relevant yet.

> `/nix/store/dc53arqhqjh32jpbnn5jdai5qqqgsbi9-gvfs-1.50.5/lib/gvfs/libgvfscommon.so: undefined symbol: g_task_set_source_tag`

These GLib-related errors are probably harmless, because we try to turn off GIO modules anyway. They are unlikely to be the root cause of Dota 2 crashing. We can try to investigate that later, as a low priority.

> `13:24:48.739187: pressure-vessel-wrap[54736]: W: "/run/current-system/sw/bin/getent" is unlikely to appear in "/run/host"`

This is *probably* harmless. What is `/run/current-system/sw/bin`? Is that a NixOS thing? If yes, we should probably make sure it's available to the SLR container. Again, this is unlikely to be what's breaking Dota 2.

> `13:24:48.740104: pressure-vessel-wrap[54736]: W: Found more than one possible libdrm data directory from provider`

This is probably harmless. On NixOS we can probably skip the code path that needs to know this, and silence the warning. I think this is also unlikely to be what's breaking Dota 2.

**Comment by smcv** [comment]
@kisak-valve or @SkyLeite: NixOS is sufficiently unusual to sometimes have its own OS-specific issues, so it might be best to retitle this to "Dota 2 on NixOS segfaults on startup using the new Sniper runtime" to separate it from any other Dota 2 reports.

**Comment by SkyLeite** [comment]
> @SkyLeite, please could you capture the information from Help -> Steam Runtime Diagnostics? (This used to be in Help -> System Information, but newer versions of Steam have moved it.)

Absolutely! Here it is: https://gist.github.com/SkyLeite/6cf088cde248c789073a2a92988227dc

I'll also amend the original post with the URL.

> This is probably harmless. What is /run/current-system/sw/bin? Is that a NixOS thing? If yes, we should probably make sure it's available to the SLR container. Again, this is unlikely to be what's breaking Dota 2.

Yes, that's a NixOS thing. `/run/current-system/sw/bin` is in my user's `PATH` and it contains symlinks to programs that are in the Nix store. It's probably meant to be an equivalent to `/usr/bin` in a "normal" Unix distro but I'm not sure.

> It would also be useful if you can try the client_beta branch of "Steam Linux Runtime - sniper". It's the same procedure as [opting in to the beta branch of a game](https://help.steampowered.com/en/faqs/view/5A86-0DF4-C59E-8C4A), but instead of editing the Properties of the game, you'd edit the Properties of the "Steam Linux Runtime - sniper" tool.

Done. Same issue, unfortunately. Here are the logs, which will also be amended to the OP: https://gist.github.com/SkyLeite/a3e679e512bfdc71c045b3543ab3ddcd

> Another useful thing to try would be launching a different title that uses sniper. At the moment there are not very many, so that basically means [Endless Sky](https://store.steampowered.com/app/404410/Endless_Sky/) or [Retroarch](https://store.steampowered.com/app/1118310/RetroArch/), which are both free, open source and not particularly big.

If those work but Dota 2 doesn't, then the crash is probably more likely to be a Dota 2 bug. If they don't work, then it's probably a problem with sniper/SLR.

Bingo! Endless Sky works perfectly (and also looks like a pretty nifty game so I'll definitely give it a try :stuck_out_tongue:). Should I close this issue and re-open it in the Dota 2 repository?

**Comment by SkyLeite** [comment]
Ah! It's also worth noting that other people in the nixpkgs repository have reported that the game works on their systems, so this doesn't seem to be widespread.

**Comment by smcv** [comment]
@kisak-valve can convert steam-runtime bugs into Dota-2 bugs, so there's no need to close and reopen a bug.

This might be specific to some particular GPUs or Vulkan drivers or something: some people in the Dota-2 issue tracker have said that Dota 2 is crashing on startup, on OSs where it otherwise works.

For instance, adding `-vulkan_disable_graphics_pipeline_library` to the end of the launch options was a workaround for some people: does that work for you?

If an option like that is necessary, then that should probably still be considered to be a Dota 2 bug, but it might help a Dota 2 developer to narrow down what is going wrong and where.
